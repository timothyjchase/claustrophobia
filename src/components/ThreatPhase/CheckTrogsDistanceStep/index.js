import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Button, Checkbox } from 'semantic-ui-react'

class CheckTrogsDistanceStep extends Component {
  constructor(props) {
    super(props)
    this.state = { trogsFar: false, trogsClose: false }
  }

  render() {
    const { completeThreatTrogsDistanceStep } = this.props
    const { trogsFar, trogsClose } = this.state

    return (
      <div>
        <Checkbox
          id="trogsFar"
          label="Any Troglogytes 2 or more tiles away from the closest human?"
          onChange={() => this.setState({ trogsFar: !trogsFar })}
          checked={trogsFar}
        />
        <br />
        <br />
        <Checkbox
          id="trogsClose"
          label="Any Troglogytes adjacent to or on the same tile as a human?"
          onChange={() => this.setState({ trogsClose: !trogsClose })}
          checked={trogsClose}
        />
        <br />
        <br />
        <Button.Group vertical fluid>
          <Button
            primary
            icon="play"
            content="Next Phase"
            onClick={() =>
              completeThreatTrogsDistanceStep(trogsFar, trogsClose)
            }
          />
        </Button.Group>
      </div>
    )
  }
}

CheckTrogsDistanceStep.propTypes = {
  completeThreatTrogsDistanceStep: PropTypes.func.isRequired,
}

export default CheckTrogsDistanceStep
