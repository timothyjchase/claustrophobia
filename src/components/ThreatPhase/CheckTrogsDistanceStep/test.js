import React from 'react'
import { Button } from 'semantic-ui-react'
import CheckTrogsDistanceStep from './index'

describe('<CheckTrogsDistanceStep />', () => {
  describe('with basic props', () => {
    const props = {
      completeThreatTrogsDistanceStep: jest.fn(),
    }
    const component = shallow(<CheckTrogsDistanceStep {...props} />)

    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
    it('trogsFar checkbox should change trogsFar', () => {
      expect(component.state().trogsFar).toEqual(false)
      component.find({ id: 'trogsFar' }).prop('onChange')()
      expect(component.state().trogsFar).toEqual(true)
    })
    it('trogsClose checkbox should change trogsClose', () => {
      expect(component.state().trogsClose).toEqual(false)
      component.find({ id: 'trogsClose' }).prop('onChange')()
      expect(component.state().trogsClose).toEqual(true)
    })
    it('button should completeThreatTrogsDistanceStep with trogsFar and trogsClose', () => {
      const { trogsFar, trogsClose } = component.state()
      component.find(Button).prop('onClick')()
      expect(props.completeThreatTrogsDistanceStep).toBeCalled()
      expect(props.completeThreatTrogsDistanceStep.mock.calls[0][0]).toBe(
        trogsFar,
      )
      expect(props.completeThreatTrogsDistanceStep.mock.calls[0][1]).toBe(
        trogsClose,
      )
    })
  })
})
