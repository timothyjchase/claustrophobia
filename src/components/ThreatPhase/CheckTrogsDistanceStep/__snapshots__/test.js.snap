// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckTrogsDistanceStep /> with basic props should match snapshot 1`] = `
<div>
  <Checkbox
    checked={false}
    id="trogsFar"
    label="Any Troglogytes 2 or more tiles away from the closest human?"
    onChange={[Function]}
    type="checkbox"
  />
  <br />
  <br />
  <Checkbox
    checked={false}
    id="trogsClose"
    label="Any Troglogytes adjacent to or on the same tile as a human?"
    onChange={[Function]}
    type="checkbox"
  />
  <br />
  <br />
  <ButtonGroup
    fluid={true}
    vertical={true}
  >
    <Button
      as="button"
      content="Next Phase"
      icon="play"
      onClick={[Function]}
      primary={true}
      role="button"
    />
  </ButtonGroup>
</div>
`;
